superset-frontend/src/components/ErrorMessage/TimeoutErrorMessage.tsx

```diff
@@ -0,0 +1,234 @@
+          target="_blank"
+        >
+          <i className="fa fa-external-link" />
+        </a>
+      </p>
+      {['chart', 'dashboard'].includes(extra.location) && extra.owners && (
+        <>
+          <br />
+          <p>{t('Please reach out to the Chart Owner(s) for assistance.')}</p>
+          <p>{t('Chart Owner(s): %s', extra.owners.join(', '))}</p>
+        </>
+      )}
+    </>
+  );
+
+  const copyText = `${title}
+${t('This may be triggered by:')}
+${ERROR_DATASOURCE_TOO_LARGE.message}
+${ERROR_DATASOURCE_UNDER_LOAD.message}`;
+
+  return (
+    <ErrorAlert>
+      <div className="topRow">
+        <LeftSideContent>
+          <Icon
+            className="icon"
+            name="error"
+            color={supersetTheme.colors.error.base}
+          />
+          <strong>{t('Timeout Error')}</strong>
+        </LeftSideContent>
+        {extra.location === 'dashboard' && (
+          <>
+            <a className="link" onClick={() => setIsModalOpen(true)}>
+              {t('See More')}
+            </a>
+            <ErrorModal show={isModalOpen} onHide={() => setIsModalOpen(false)}>
+              <Modal.Header className="header">
+                <LeftSideContent>
+                  <Icon
+                    className="icon"
+                    name="error"
+                    color={supersetTheme.colors.error.base}
+                  />
+                  <div className="title">{t('Timeout Error')}</div>
+                </LeftSideContent>
+                <span
+                  role="button"
+                  tabIndex={0}
+                  onClick={() => setIsModalOpen(false)}
+                >
+                  <Icon name="close" />
+                </span>
+              </Modal.Header>
+              <Modal.Body>
+                <p>{title}</p>
+                <br />
+                {message}
+              </Modal.Body>
+              <Modal.Footer>
+                <CopyToClipboard
+                  text={copyText}
+                  shouldShowText={false}
+                  wrapped={false}
+                  copyNode={<Button onClick={noOp}>{t('Copy Message')}</Button>}
+                />
+                <Button bsStyle="primary" onClick={() => setIsModalOpen(false)}>
+                  {t('Close')}
+                </Button>
+              </Modal.Footer>
+            </ErrorModal>
+          </>
+        )}
+      </div>
+      {['chart', 'sqllab'].includes(extra.location) && (
+        <div className="errorBody">
+          <p>{title}</p>
+          {!isMessageExpanded && (
+            <a className="link" onClick={() => setIsMessageExpanded(true)}>
+              {t('See More')}
+            </a>
+          )}
+          {isMessageExpanded && (
+            <>
+              <br />
+              {message}
+              <a className="link" onClick={() => setIsMessageExpanded(false)}>
+                {t('See Less')}
+              </a>
+            </>
+          )}
+        </div>
+      )}
```

Comment:
Instead of two big code chunks, how about creating two separate subcomponent `SimpleTimeoutErrorMessage` and `DetailedTimeoutErrorMessage`?

Reply:
Let me look into simplifying this a bit more. I might pull some things out into smaller, more reusable components. It's a bit tricky, because depending on where the error is being rendered, it's supposed to look different (sometimes using a modal, sometimes expanding inline)